File: 485.js
var findMaxConsecutiveOnes = function(nums) {
    //if the element at a given index is 1, increase count
    //while increasing the count, also make sure to check in the max and update it
    //
    //if the element at a given index is 0, then reset the count
    //make sure the count > 0 to reset it, to minimize the double work.
    let count = 0;
    let max = 0;
    for(let i = 0; i < nums.length; i++){
        if(nums[i] === 1){
            count++;
            if(max < count){
                max = count;
            }
        }else{
            if(count > 0){
                count = 0;
            }
        }
    }
    return max;
};

File: 2021 LeetCode/485.java
// 485. Max Consecutive Ones

class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int i = 0;
        for(int j = 0; j < nums.length; j++) {
            if(nums[j] == 1) {
                i++;
            } else {
                max = Math.max(max, i);
                i = 0;
            }
        }
        // if(nums.length >= 1 && nums[nums.length-1] == 1) 
        //     max = Math.max(max,i);
        return Math.max(max,i);
    }
}